testing_and_setup/compass/ocean/scripts/vertical_grid/make_vertical_grid.py

```diff
@@ -0,0 +1,198 @@
+
+    layerThickness = np.zeros(nz)
+    dz = [epsilon]
+    z = [0]
+    count = 0
+
+    while nLayers != nz and count < maxit:
+        zval = -epsilon
+        dz = [epsilon]
+        z = [0]
+        nLayers = 0
+        while zval > -Hmax:
+            difference = dz_z(zval, Hmax, epsilon, dz2) - zval
+            while abs(difference) > 0.3:
+                zval -= epsilon
+                difference = dz_z(zval, Hmax, epsilon, dz2) - z[nLayers] + zval
+            z.append(zval)
+            dz.append(dz_z(zval, Hmax, epsilon, dz2))
+            nLayers += 1
+            zval -= epsilon
+
+        dz_arr = np.asarray(dz)
+        ind = abs(dz_arr - dz1).argmin()
+
+        dztemp = dz_arr[ind:]
+        nLayers = len(dztemp)
+        change = nz - nLayers
+        dz2 -= float(change)
+        count += 1
+
+    layerThickness[:nLayers] = dztemp
+    nVertLevels = nz
+    botDepth = np.zeros(nVertLevels)
+    midDepth = np.zeros(nVertLevels)
+    botDepth[0] = layerThickness[0]
+    midDepth[0] = 0.5 * layerThickness[0]
+
+    for i in range(1, nVertLevels):
+        botDepth[i] = botDepth[i - 1] + layerThickness[i]
+        midDepth[i] = midDepth[i - 1] + 0.5 * \
+            (layerThickness[i] + layerThickness[i - 1])
+
+    if count >= maxit:
+        print('Error: grid did not converge, adjust parameters')
+    else:
+        refBottomDepth[:] = botDepth
+        refMidDepth[:] = midDepth
+        refLayerThickness[:] = layerThickness[:nVertLevels]
+        ncfile.close()
+
+    if plot_vertical_grid:
+        import matplotlib
+        import matplotlib.pyplot as plt
+        matplotlib.use('Agg')
+        fig = plt.figure()
+        fig.set_size_inches(16.0, 8.0)
+        zInd = np.arange(1, nVertLevels + 1)
+        plt.clf()
+
+        plt.subplot(2, 2, 1)
+        plt.plot(zInd, midDepth, '.')
+        plt.gca().invert_yaxis()
+        plt.xlabel('vertical index (one-based)')
+        plt.ylabel('layer mid-depth [m]')
+        plt.grid()
+
+        plt.subplot(2, 2, 2)
+        plt.plot(layerThickness, midDepth, '.')
+        plt.gca().invert_yaxis()
+        plt.xlabel('layer thickness [m]')
+        plt.ylabel('layer mid-depth [m]')
+        plt.grid()
+
+        plt.subplot(2, 2, 3)
+        plt.plot(zInd, layerThickness, '.')
+        plt.xlabel('vertical index (one-based)')
+        plt.ylabel('layer thickness [m]')
+        plt.grid()
+
+        txt = \
+            'number layers: ' + str(nz) + '\n' + \
+            'bottom depth requested:  ' + '{:8.2f}'.format(bottom_depth) + '\n' +  \
+            'bottom depth actual:     ' + '{:8.2f}'.format(np.amax(botDepth)) + '\n' +  \
+            'min thickness reqeusted: ' + '{:8.2f}'.format(dz1_in) + '\n' + \
+            'min thickness actual:    ' + '{:8.2f}'.format(np.amin(layerThickness[:])) + '\n' + \
+            'max thickness reqeusted: ' + '{:8.2f}'.format(dz2_in) + '\n' + \
+            'max thickness actual:    ' + '{:8.2f}'.format(np.amax(layerThickness[:]))
```

Comment:
This is a strange combination for formatting strings.  Generally, `str` addition is not used, and it isn't needed at all when you're using `format` like here.

Reply:
I can replace the str(nz) with something like `'{:d}'.format(nz)`. Is that what you mean? Or was there something else?

Reply:
In this particular case, the pluses might be okay for each line.  In general, there would only be one string with a bunch of curly brackets and one format statement at the end for all the variables:

```python
         txt = 'number layers: {} \n' \
               'bottom depth requested:  {:8.2f}\n'  \
               'bottom depth actual:     {:8.2f}\n'.format(nz, bottom_depth, np.amax(botDepth))
```
In this case, since it's so long, maybe this would work (i.e. a plus sign per line):
```python
        txt = 'number layers: {} \n'.format(nz) + \
              'bottom depth requested:  {:8.2f}\n'.format(bottom_depth) +  \
              'bottom depth actual:     {:8.2f}\n'.format(np.amax(botDepth)) +  \
...
```

Reply:
In general, we don't use pluses with strings in python.